pkg/operation/botanist/worker.go

```diff
@@ -137,16 +137,9 @@ func CloudConfigUpdatedForAllWorkerPools(workers []gardencorev1beta1.Worker, wor
 	return result
 }
 
-const (
-	// MCMPreferNoScheduleKey is used to identify machineSet nodes on which PreferNoSchedule taint is added on
-	// older machineSets during a rolling update
-	MCMPreferNoScheduleKey = "deployment.machine.sapcloud.io/prefer-no-schedule"
-)
-
-// nodeToBeDeleted checks if the MCM has set the node to be deleted.
-func nodeToBeDeleted(node corev1.Node) bool {
-	for _, taint := range node.Spec.Taints {
-		if taint.Key == MCMPreferNoScheduleKey && taint.Effect == corev1.TaintEffectPreferNoSchedule {
```

Comment:
Is there a test case I can add somewhere that will help us find out what's necessary here? This might also help with the above question of "does this really solve the issue".